### DISCUSSION TITLE
AES-256- GCM encryption iOS not working

### DISCUSSION THREAD

**Discussion by arindamSantraAECB** [krzyzanowskim/CryptoSwift#1009] [Q&A]
For iOS AES-256- GCM encryption not working.  PLEASE CHECK MY CODE . help me to resolve this 

func getencreptedtext256(text:String) -> String {
        
        // encryption part begins
        let digest = SHA2(variant: SHA2.Variant.sha256)
        let encryptedKeyText = "TEST"
        let shaFull = digest.calculate(for: Array(encryptedKeyText.utf8))
        let shaKey = copyOfRange(arr: shaFull, from: 0, to: 16)!
        let IV = copyOfRange(arr: shaFull, from: 0, to: 12)!
        let gcm = GCM(iv: iv, mode: .combined)
                
        let encrypted: Array<UInt8> = try! AES(
            key: shaKey,
            blockMode: gcm,
            padding: .noPadding
        )
            .encrypt(text.bytes)
        let encryptedText = removeWhiteSpaces(text: encrypted.toBase64())
        print("enc base64: \(encryptedText)")
        
        return encryptedText
        
        
    }

**Comment by krzyzanowskim** [comment]
What exactly do you mean by "not working"?
